### DISCUSSION TITLE
Meeting Address Changes

### DISCUSSION THREAD

**Discussion by egbdaa** [code4recovery/12-step-meeting-list#901] [Q&A]
Hi I'm very new to this and rookie. Using Wordpress for our website egbdaa.org. I have 2 issues 1. I can't get it to accept the correct meeting postal code. When I type in L9M 1H1 click update it changes to L9M 1P2 everything else stays as it should be.  2. When I go to the "Meeting Guide" the address has changed to 95 Robert Street West it should be East as it is in the website. The complete correct address "Monday Night Serenity Group, Covenant Christian Community Chruch, 95 Robert Street East Penetanguishene Ontario L9M 1H1 We are part of Eastern Georgian Bay District 7, Western Ontario General Service Area 86 Canada HELP P.S. I have sent a report to Google Maps as well with no responce. Thanks Andy

**Comment by joshreisner** [comment]
yep sorry, this is an odd one! what's happening is we use google to geocode addresses. if you give the google geocoding api 

`95 Robert St E, Penetanguishene, ON L9M 1H1, Canada` it comes back with 
`95 Robert St E, Penetanguishene, ON L9M 1P2, Canada` which is what you see on your website. (at least the pin is in the right place!)

but then when it goes to the app
`95 Robert St E, Penetanguishene, ON L9M 1P2, Canada` gets geocoded and comes back with 
`95 Robert St W, Penetanguishene, ON L9M 1P2, Canada` which is wrong 😞 

do you have access to your theme's `functions.php`? we recommend adding:

```
if (function_exists('tsml_custom_addresses')) {
    tsml_custom_addresses(array(
        '95 Robert St E, Penetanguishene, ON L9M 1H1, Canada' => array(
            'formatted_address' => '95 Robert St E, Penetanguishene, ON L9M 1H1, Canada',
            'city' => 'Penetanguishene',
            'latitude' => 44.7744902,
            'longitude' => -79.9223015,
            'approximate' => 'no',
        ),
    ));
}
```

**Reply by egbdaa** [reply]
Hi Josh. Thanks for the reply, but I have no idea what functions.php is or where it is. I'm just trying to maintain and keep our website up to date. Can you or ??? correct this? Thanks Andy

**Reply by tim-rohrer** [reply]
And @egbdaa, as a new webservant, may I suggest signing up for and attending [NAATW](https://naatw.org)? It will be held in Seattle the weekend after next, but you can also attend online. Great way to connect with other tech servants and learn how to do some of these things. 

[TIAA Forum](https://tiaa-forum.org) is another good resource you should check out.

**Reply by josephrwilcox** [reply]
> yep sorry, this is an odd one! what's happening is we use google to geocode addresses. if you give the google geocoding api
> 
> `95 Robert St E, Penetanguishene, ON L9M 1H1, Canada` it comes back with `95 Robert St E, Penetanguishene, ON L9M 1P2, Canada` which is what you see on your website. (at least the pin is in the right place!)
> 
> but then when it goes to the app `95 Robert St E, Penetanguishene, ON L9M 1P2, Canada` gets geocoded and comes back with `95 Robert St W, Penetanguishene, ON L9M 1P2, Canada` which is wrong 😞
> 
> do you have access to your theme's `functions.php`? we recommend adding:
> 
> ```
> if (function_exists('tsml_custom_addresses')) {
>     tsml_custom_addresses(array(
>         '95 Robert St E, Penetanguishene, ON L9M 1H1, Canada' => array(
>             'formatted_address' => '95 Robert St E, Penetanguishene, ON L9M 1H1, Canada'
>             'city' => 'Penetanguishene',
>             'latitude' => 44.7744902,
>             'longitude' => -79.9223015,
>             'approximate' => 'no',
>         ),
>     ));
> }
> ```

I think you're missing a comma after 'formatted_address' => '95 Robert St E, Penetanguishene, ON L9M 1H1, Canada'

Like the original poster, I'm not overly technical and updated my functions.php for the theme in my File Manager, so I kind of blew up my site for little while.

I did end up pasting the following into my functions.php using the theme editor (after backing up the original function.php file) and it worked like a charm...thank for you this:

if (function_exists('tsml_custom_addresses')) {
    tsml_custom_addresses(array(
        '2200 State Hill Road, Wyomissing, PA 19610, USA' => array(
            'formatted_address' => '2200 State Hill Road, Wyomissing, PA 19610, USA',
            'city' => 'Wyomissing',
            'latitude' => 40.339000,
            'longitude' => -75.980560,
            'approximate' => 'no',
        ),
    ));
}

**Reply by joshreisner** [reply]
oops yes you're right, sorry about that! (i've updated the code sample now)

**Comment by 97eighty** [comment]
Hey @egbdaa, Glad to help. The functions.php is located in your themes folder. Pasting the code in from above at the bottom of the file should work for you. There are two methods to access your functions.php file. If you are using a [Block theme]), both the Theme and Plugin File Editor will be listed under Tools. If you are using a Classic theme, the Theme File Editor will be listed under Appearance and the Plugin File Editor will be listed under Plugins. Or, if these are not available (They can be turned off in some installations) you can access the functions.php through FTP. Do you have this sort of access?

**Reply by tim-rohrer** [reply]
It should be added that if you add this code to the regular theme `functions.php`, the changes will be overwritten the next time the theme is updated. It is generally a good practice to add a child theme and make the changes there. 

@97eighty's suggestion of using the Theme File Editor is not something I've not worked on before, so perhaps that saves the additional code is such a way it is not overwritten?
